cmake/Modules/FindPoco.cmake

```diff
@@ -118,6 +118,35 @@ endif()
 # Assume we didn't find it.
 set(Poco_FOUND 0)
 
+# check if found Poco version is at least version 1.4.1p1
+# since older versions don't use recursive mutexes (on Linux)
+if(Poco_INCLUDE_DIR)
+  find_file (
+      Poco_VERSION_FILE NAMES "Version.h"
+      PATHS "${Poco_INCLUDE_DIR}"
+      PATH_SUFFIXES "Poco" "Foundation/include/Poco"
+      DOC "Path of Poco/Version.h file"
+    NO_DEFAULT_PATH
+  )
+  if(NOT Poco_VERSION_FILE)
+    message(STATUS "Found Poco version (< 1.4.0) is too old, building from source instead")
```

Comment:
Looks like this version file was [added in 1.3.7](https://github.com/pocoproject/poco/commit/7a563faa29be8dd5a47e4ea2f5aa024e0e27bbcc#diff-a240205fff6d12b8a861b5cc16475a65), can you clarify how you picked "1.4.0" for the printed message?

Reply:
The `CHANGELOG` lists the addition in the "Release 1.4.0" section. That is where I got the version number from. Also I don't see any listing of a version 1.3.7 (neither in the changelog nor as a tag).

Reply:
ok, I guess they had a branch for it but didnt ever released 1.3.7 and several commits on that branch say "preparing for 1.4.0". Sorry for the noise